tests/5.0/requires/test_requires_unified_shared_memory.F90

```diff
@@ -0,0 +1,46 @@
+!===--- test_requires_unified_shared_memory.F90 ----------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test just uses requries unified_shared_memory in a simple target region
+!  and it is intended to show that this would not break anything in the compiler
+!
+!===------------------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+PROGRAM test_requires_unified_shared_memory
+   USE iso_fortran_env
+   USE ompvv_lib
+   USE omp_lib
+   implicit none
+   INTEGER:: errors, test_var
+   LOGICAL:: isOffloading
+
+!$omp requires unified_shared_memory
+
+   OMPVV_TEST_AND_SET_OFFLOADING(isOffloading)
+
+   OMPVV_WARNING_IF(isOffloading .eqv. .false., "With no offloading, unified shared memory is guaranteed due to host execution")
+
+   errors = 0
+
+   test_var = 0
+
+   OMPVV_INFOMSG("Unified shared memory testing")
+
+!$omp target map(tofrom: test_var)
```

Comment:
Except for `!$omp requires unified_shared_memory` this does not really check for unified shared memory.
Even `!$omp target map(tofrom: test_var)` maps the variable explicitly.

With implicit mapping it would be at least `firstprivate` as `test_var` is a scalar (both in the OpenMP and in the Fortran sense).

Thus, I would remove at least the explicit mapping. A more thorough test would probably involve pointers as – at least for `type(c_ptr)` cannot then see what has to be mapped.

_Side note: gfortran of GCC 11 now also parses 'requires' but except for the atomic settings, it is unimplemented and shows similar to GCC's C/C++ compiler: `Error: Sorry, ‘unified_shared_memory’ clause at (1) on REQUIRES directive is not yet supported`_

Reply:
@tob2 this is a basic test, it does not check working of the pragma, just support. Essentially that the test compiles correctly thus verifying support. We will be working on more elaborate tests as well.